src/util/namespace.h

```diff
@@ -84,8 +84,20 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`. If a symbol is
+  /// found it will be returned in \p symbol.
+  /// \return False iff the requested symbol is found in at least one of the two
+  /// tables
```

Comment:
Continuing the discussion we had on the earlier versions: as first order of business the audience will need to understand the purpose of the function, much less so the implementation. Consequently I would prefer if all the documentation were placed with `namespace_baset`, which is the API. I'd suggest:
"Search for a symbol named \p name; iff found, set \p symbol to point to the symbol and return false; else \p symbol is unmodified and true is returned. With multiple symbol tables, `symbol_table1` is searched first and then `symbol_table2`. \return False ... [as above]"